go/runtime/client/client.go

```diff
@@ -93,16 +93,34 @@ func (c *runtimeClient) doSubmitTxToLeader(
 	defer close(submitCtx.closeCh)
 
 	op := func() error {
-		_, err := client.SubmitTx(submitCtx.ctx, req)
+		// Populate request with the current epoch.
+		// TODO: Is this the right way to fetch current epoch?
+		epochTime, err := c.common.consensus.EpochTime().GetEpoch(context.Background(), consensus.HeightLatest)
```

Comment:
This is also not the right way to fetch the client's current epoch as you need to check what epoch the client block watcher has actually processed. This information should be made available in the `blockWatcher` (updated in `watch()` when a block kind is `block.EpochTransition`). The watcher instance is available in `SubmitTx` which calls `doSubmitTxToLoader`.